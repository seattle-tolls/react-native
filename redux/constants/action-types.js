export const SET_TOLL_INFO = 'SET_TOLL_INFO'
export const FETCH_TOLL_INFO = 'FETCH_TOLL_INFO'

export const SET_TOLL_NAME = 'SET_TOLL_NAME'