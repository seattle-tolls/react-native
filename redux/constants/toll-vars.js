export const TOLL_99 = 'toll99'
export const TOLL_520 = 'toll520'