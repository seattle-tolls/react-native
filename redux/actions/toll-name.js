import { SWITCH_TOLL_NAME } from '../constants/action-types'

export const switchTollName = (payload) => ({
  type: SWITCH_TOLL_NAME, payload,
})