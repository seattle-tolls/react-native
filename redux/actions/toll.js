import { SET_TOLL_INFO, FETCH_TOLL_INFO } from '../constants/action-types'

export const setTollInfo = (payload) => ({
  type: SET_TOLL_INFO, payload,
})

export const fetchTollInfo = () => ({
  type: FETCH_TOLL_INFO,
})