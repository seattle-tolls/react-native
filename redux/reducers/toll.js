import { SET_TOLL_INFO } from '../constants/action-types'
import { createSelector } from 'reselect'

const initialState = {}

const tollReducer = (toll = initialState, action) => {
  switch (action.type) {
    case SET_TOLL_INFO:
      return action.payload
    default:
      return toll
  }
}

export const tollInfo = (state) => {
  const tollInfoObj = state.toll[state.tollName]
  console.log('TOLL INFO', tollInfoObj && tollInfoObj.name)
  return tollInfoObj
}

// FIXME: fix the other selectors according with the derived data

// export const todayInfo = state => state.toll.todayInfo
// export const todaySchedule = state => state.toll.todaySchedule

// export const todayScheduleAsArray = createSelector(
//   todaySchedule,
//   (schedule) => {
//     if(schedule[0]){
//       const scheduleArray = Object.keys(schedule).map(hour => {
//         return ({
//           id: hour,
//           goodToGo: schedule[hour].goodToGo,
//           payByMail: schedule[hour].payByMail,
//           schedule: schedule[hour].schedule,
//         })
//       })
//       return scheduleArray
//     }
//   }
// )

// export const currentTimeInfo = createSelector(
//   todayInfo,
//   todaySchedule,
//   (info, schedule) => {
//     for(let i = info.hour; i >= 0; i--){
//       if(schedule[i]){
//         return{
//           dayName: info.dayName,
//           holidayName: info.holidayName,
//           ...schedule[i],
//         }
//       }
//     }
//   }
// )

export default tollReducer
