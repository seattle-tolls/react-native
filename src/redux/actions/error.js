import { REPORT_ERROR } from '../constants/action-types'

export const reportError = (error) => ({
  type: REPORT_ERROR, error,
})