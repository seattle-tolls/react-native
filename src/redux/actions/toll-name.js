import { SET_TOLL_NAME } from '../constants/action-types'

export const setTollName = (payload) => ({
  type: SET_TOLL_NAME, payload,
})
