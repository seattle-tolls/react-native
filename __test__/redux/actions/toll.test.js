import { fetchTollInfo, setTollInfo } from '../../../src/redux/actions/toll'
import { SET_TOLL_INFO, FETCH_TOLL_INFO } from '../../../src/redux/constants/action-types'

describe('ACTIONS', () => {
  describe('actions/toll', () => {
    it('should run fetchTollInfo', () => {
      const fetchToll = fetchTollInfo()
      expect(fetchToll.type).toEqual(FETCH_TOLL_INFO)
      expect(fetchToll).toMatchSnapshot()
    })

    const tollInfo = [
      { toll99: { name: '99 tunnel' } },
      { toll520: { name: '520 toll bridge' } },
      { fakeToll: { name: 'fake toll name' } },
    ]

    tollInfo.forEach(payload => {
      it(`should run setTollInfo on the '${Object.keys(payload)[0]}'`, () => {
        const setToll = setTollInfo(payload)
        expect(setToll.type).toEqual(SET_TOLL_INFO)
        expect(setToll.payload).toEqual(payload)
        expect(setToll).toMatchSnapshot()
      })
    })
  })
})