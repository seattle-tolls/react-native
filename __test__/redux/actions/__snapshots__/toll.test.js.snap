// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ACTIONS actions/toll should run fetchTollInfo 1`] = `
Object {
  "type": "FETCH_TOLL_INFO",
}
`;

exports[`ACTIONS actions/toll should run setTollInfo on the 'fakeToll' 1`] = `
Object {
  "payload": Object {
    "fakeToll": Object {
      "name": "fake toll name",
    },
  },
  "type": "SET_TOLL_INFO",
}
`;

exports[`ACTIONS actions/toll should run setTollInfo on the 'toll99' 1`] = `
Object {
  "payload": Object {
    "toll99": Object {
      "name": "99 tunnel",
    },
  },
  "type": "SET_TOLL_INFO",
}
`;

exports[`ACTIONS actions/toll should run setTollInfo on the 'toll520' 1`] = `
Object {
  "payload": Object {
    "toll520": Object {
      "name": "520 toll bridge",
    },
  },
  "type": "SET_TOLL_INFO",
}
`;
